src/mongoose_tcp_listener.erl

```diff
@@ -115,34 +115,54 @@ accept_loop(ListenSocket, Module, Opts = #{proxy_protocol := ProxyProtocol}) ->
             ?MODULE:accept_loop(ListenSocket, Module, Opts)
     end.
 
--spec do_accept(gen_tcp:socket(), boolean()) ->
+-spec do_accept(gen_tcp:socket(), options()) ->
     {ok, gen_tcp:socket(), connection_details()} | {error, term()}.
-do_accept(ListenSocket, ProxyProtocol) ->
+do_accept(ListenSocket, Opts) ->
     case gen_tcp:accept(ListenSocket) of
-        {ok, Socket} when ProxyProtocol ->
-            read_proxy_header(Socket);
         {ok, Socket} ->
-            {ok, {DestAddr, DestPort}} = inet:sockname(Socket),
-            {ok, {SrcAddr, SrcPort}} = inet:peername(Socket),
+            ?MODULE:read_connection_details(Socket, Opts);
+        Other ->
+            Other
+    end.
+
+-spec read_connection_details(gen_tcp:socket(), options()) ->
+    {ok, gen_tcp:socket(), connection_details()} | {error, term()}.
+read_connection_details(Socket, #{proxy_protocol := true}) ->
+    read_proxy_header(Socket);
+read_connection_details(Socket, _Opts) ->
+    case {inet:sockname(Socket), inet:peername(Socket)} of
+        {{ok, {DestAddr, DestPort}}, {ok, {SrcAddr, SrcPort}}} ->
             {ok, Socket, #{proxy => false,
                            src_address => SrcAddr,
                            src_port => SrcPort,
                            dest_address => DestAddr,
                            dest_port => DestPort}};
         Other ->
-            Other
+            gen_tcp:close(Socket),
+            {error, simple_reason(Other)}
```

Comment:
I wonder why CodeCov doesn't recognize any of these error handling cases as covered 🤔

Reply:
because we mock the file under the test, mocked files don't collect coverage info.

Reply:
yea, the test runs :)
https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3695/127964/riak_mnesia.24.3.4/big/ct_run.test%4086c1dd277172.2022-06-26_19.48.53/big_tests.tests.tcp_listener_SUITE.logs/run.2022-06-26_20.05.10/suite.log.html

Reply:
Ok, thx for the explanation :)